File: 1768_Merge_Srings_Alternately/main.js
/**
 * @param {string} word1
 * @param {string} word2
 * @return {string}
 */
var mergeAlternately = function(word1, word2) {
    const min_size = Math.min(word1.length,word2.length);
    const new_word =[];
    for(let i=0; i<min_size; i++){
        new_word.push(word1[i]);
        new_word.push(word2[i]);
    }

    if (word1.length < word2.length){
        new_word.push(word2.slice(min_size))
    }

    if (word1.length > word2.length){
        new_word.push(word1.slice(min_size))
    }

    return new_word.join("")
};

File: 1768_Merge_Srings_Alternately/main.py
class Solution:
    def mergeAlternately(self, word1: str, word2: str) -> str:
        """
        Since this problem relys on the minimum length of a word we can solve this by precomputing the minimum.
        """

        min_size = min(len(word1),len(word2))
        new_word = []
        for i in range(min_size):
            new_word.append(word1[i])
            new_word.append(word2[i])    
        if len(word1) > len(word2):
            new_word.append(word1[min_size:])
        if len(word1) < len(word2):
            new_word.append(word2[min_size:])
        return "".join(new_word)
